docs/docs/guides/atom-effects.md

```diff
@@ -239,6 +241,89 @@ const currentUserIDState = atom({
 });
 ```
 
+## Asynchronous Storage Persistence
+
+If your persisted data needs to be retrieved asynchronously, you can either use a [`Promise`](#promise) in the `setSelf()` function or create an [asynchronous](#asynchronous) function and call that.
+
+### `Promise`
+
+By synchronously calling `setSelf()` with a `Promise` you'll be able to wrap the components inside of the `<RecoilRoot/>` with a `<Suspense/>` component to show a fallback while waiting for `Recoil` to load the persisted values.
+
+The `<Suspense/>` fallback will be shown until the `setSelf()` function has been called.
+
+Note that if the `atoms` later are "reset", they will revert to their default value, and not the initialized value.
+
+```jsx
+const localStorageEffect = key => ({setSelf, onSet}) => {
+  setSelf(
+    localStorage.getItem(key).then((savedValue) => {
```

Comment:
`localStorage` only stores UTF-16.  Best to avoid `localStorage` completely for the async examples.  You could also just use `localForage` for this:

```
setSelf(localForage.getItem(key).then(savedValue =>
  savedValue != null
    ? JSON.parse(savedValue)
    : new DefaultValue() // Abort initialization if no value was stored
));
```

Reply:
Sorry, that's my bad.
We already talked about this, so simply a typo.
Thanks!

Reply:
Resolved.